### PR TITLE
Add read the docs configuration

### PR DESCRIPTION
**Pull request by llawall**
A few warnings in the output that ought to be cleaned up but nothing too serious. Can be done as part of pull request to actually write some reasonable documentation - currently this just outputs an API reference section.

### CODE REVIEW

**Comment by llawall** [comment]
This pull request addresses issues #13

File: docs/conf.py [hunk 1]
```diff
@@ -55,10 +74,8 @@
 # built documents.
 #
 
-from pkg_resources import get_distribution
-release = get_distribution('winton_kafka_streams').version
-# for example take major/minor
-version = '.'.join(release.split('.')[:2])
+from setuptools_scm import get_version
+version = release = get_version(root='..')
```
Comment (satlank) [inline docs/conf.py]: Nice, setuptools_scm is very convenient!

Comment (llawall) [reply docs/conf.py]: Yes, though strictly speaking this change is against their usage advice. See the section “Usage from Sphinx” here: https://github.com/pypa/setuptools_scm/blob/master/README.rst 

The problem is that we are not either doing an editable or real install with pip as the Confluent Kafka package must be built and that doesn’t work cleanly on RTD (this should also help explain why the Kafka related modules are mocked above).

**Comment by ah-** [comment]
Nice! Is this visible already somewhere?

That travis error on nighly python 3.7
```
../../../virtualenv/python3.7-dev/lib/python3.7/site-packages/_pytest/python.py:403: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
../../../virtualenv/python3.7-dev/lib/python3.7/site-packages/py/_path/local.py:668: in pyimport
    __import__(modname)
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:955: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:656: in _load_unlocked
    ???
<frozen importlib._bootstrap>:626: in _load_backward_compatible
    ???
../../../virtualenv/python3.7-dev/lib/python3.7/site-packages/_pytest/assertion/rewrite.py:213: in load_module
    py.builtin.exec_(co, mod.__dict__)
tests/processor/serde/test_serialisation.py:1: in <module>
    from winton_kafka_streams.processor.serialization.serdes import *
winton_kafka_streams/processor/__init__.py:10: in <module>
    from ._stream_thread import StreamThread
winton_kafka_streams/processor/_stream_thread.py:14: in <module>
    from ._stream_task import StreamTask
E     File "/home/travis/build/wintoncode/winton-kafka-streams/winton_kafka_streams/processor/_stream_task.py", line 162
E       self.consumer.commit(offsets=offsets_to_commit, async=False)
E                                                           ^
E   SyntaxError: invalid syntax
```
might be because `async` is becoming a keyword? Looks more like a confluent kafka python bug.

**Comment by llawall** [comment]
I'll merge this now. As noted by @ah-  on confluentinc/confluent-kafka-python#21 the Python nightly failure is due to a planned Python language change in 3.7. We will move to using asynchronous keyword arg when available in confluent kafka Python release. 

PEP 492 transition plan for await/async to become full keywords: https://www.python.org/dev/peps/pep-0492/#deprecation-plans

Recent CPython pull request implementing this:
python/cpython#1669

**Comment by llawall** [comment]
Documentation for latest (i.e. master) build now available here: http://winton-kafka-streams.readthedocs.io/en/latest/

**Pull request merged**